metricflow/model/data_warehouse_model_validator.py

```diff
@@ -0,0 +1,120 @@
+from collections import OrderedDict
+from dataclasses import dataclass, field
+from time import perf_counter
+from typing import List, Optional, OrderedDict as ODType
+
+from metricflow.cli.cli_context import CLIContext
+from metricflow.model.objects.data_source import DataSource
+from metricflow.model.objects.user_configured_model import UserConfiguredModel
+from metricflow.model.validations.validator_helpers import ValidationError, ValidationIssue, ValidationWarning
+
+
+@dataclass
+class DataWarehouseValidationTask:
+    """Dataclass for defining a task to be used in the DataWarehouseModelValidator"""
+
+    query_string: str
+    error_message: str
+    object_ref: ODType = field(default_factory=lambda: OrderedDict())
+
+
+class DataWarehouseTaskBuilder:
+    """Task builder for standard data warehouse validation tasks"""
+
+    QUERY_TEMPLATE = "SELECT {columns_select} FROM {data_source}"
+    PARTITION_TEMPLATE = " WHERE {partition_col} IS NOT NULL"
+    WRAPPED_COL_TEMPLATE = "({column}) as col{unique_id}"
+
+    @staticmethod
+    def _wrap_col(col: str, id: int) -> str:  # noqa: D
+        return DataWarehouseTaskBuilder.WRAPPED_COL_TEMPLATE.format(column=col, unique_id=id)
+
+    @staticmethod
+    def _gen_query(data_source: DataSource, columns: List[str] = ["true"]) -> str:  # noqa: D
+        data_source_str = data_source.sql_table if data_source.sql_table else f"({data_source.sql_query})"
+
+        wrapped_cols = [DataWarehouseTaskBuilder._wrap_col(col, index) for index, col in enumerate(columns)]
+        columns_select = ", ".join(wrapped_cols)
+
+        query = DataWarehouseTaskBuilder.QUERY_TEMPLATE.format(
+            data_source=data_source_str, columns_select=columns_select
+        )
+
+        # If the data source has a partition, we need to include it in the
+        # query. This is because some data warehouses require the partition as
+        # part of the query for any table that has a partition defined
+        if data_source.partition:
```

Comment:
What happens if there are multiple partitions?

Reply:
What happens on the data warehouse side or what happens here when generating the query? 

My understanding is that the general rule on the data warehouse side is that one partition must be supplied (not all). Will that cover every data warehouse? Probably not, but time will tell and we'll tackle that when it comes up.

As far as generating the query, `data_source.partition` will only ever return one partition. So the query will be generated with one `WHERE` clause selection.

Reply:
I was asking what happens if there are multiple partition dimensions defined in the data source. `data_source.partition` will return one partition, but that's only if the data source has been defined with one partition. From looking at the implementation and usages of that call, it seems like a legacy carry over.

There's also a difference between semantic partitions and physical partitions, but resolving that difference might be out of scope for this PR.

Reply:
Paul and I talked offline about this. Semantic partitions and data warehouse partitions might not always line up. I.e. a semantic partition might not actually be a partition in the data warehouse. It isn't guaranteed that any partition defined in the configs is a partition in the data warehouse. So our best bet is to include any dimension defined as a partition for a data source in the WHERE clause and hope one of them is a partition in the data warehouse.